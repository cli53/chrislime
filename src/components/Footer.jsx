import React from 'react';
import PropTypes from 'prop-types';

export default function Footer({ contactVis, contactBar }) {
  return (
    <footer>
      <ul id='footer-links'>
        <li>linkedin</li>
        <li>Github</li>
        <li>Ask me your questions!</li>
      </ul>
      <p id='copyright'>&copy; Chris Li. All rights reserved</p>
    </footer>
  );
}

Footer.propTypes = {
  contactVis: PropTypes.bool,
  contactBar: PropTypes.func,
};


{/* <div id="contact-contain" onClick={contactBar} className={contactVis ? 'slideOut' : 'slideIn'}>
<p id="contact">Contact</p>
<div id="info-box">
  <p>LINKEDIN:<span className="contacts-info"><a href="https://www.linkedin.com/in/chrisdigitiali/" target="_blank">CHRISDIGITALI</a></span></p>
  <p>EMAIL: <span className="contacts-info"><a href="mailto:chrisli@codesmith.io?Subject=Hi%20Chris!">chrisli@codesmith.io</a></span></p>
</div>
</div> */}