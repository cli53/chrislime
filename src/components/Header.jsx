import React from 'react';
import { NavLink } from 'react-router-dom';

export default function Nav() {
  return [
    <ul id='nav-contain'>
      <li className='nav' key={1}>
        <NavLink className='nav' exact={true} to='/'>Logo</NavLink>
      </li>
      <li className='nav' key={2}>
        <NavLink className='nav' exact={true} to='/About'>About</NavLink>
      </li>
    </ul>
  ]
}
