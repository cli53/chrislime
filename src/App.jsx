import React, { Component } from 'react';
import Routes from './components/routes.jsx';
import Footer from './components/Footer.jsx';
import Header from './components/Header.jsx';


class App extends Component {
  constructor(props) {
    super(props);
    // console.log('inside THIS:', this)
    // "this" keyword is referring to the constructor function object so it holds the methods within the class, props, state object with it's properties, ref, reactInternal properties, and the updater obj which is based on the scheduler
    // THIS: Object { props: {}, context: undefined, refs: {}, updater: {…}, giphyCall: u/this.giphyCall(), giphyChange: u/this.giphyChange(), contactBar: u/this.contactBar() }
    this.state = {
      contactVisible: false,
      // giphyArr: [],
      // gipyInput: '',
    };

    // `.bind` doesn’t modify the original function, it just returns a new function with the given context whenever it is called.
    // The key here is that the constructor is only called once, not on every render. As oppose to binding the event handlers in the callbacks within the render function which will create a new function on every render and React renders often.
    // this.contactBar = this.contactBar.bind(this); // binding 'this' context of the constructor to the our methods
    // this.giphyCall = this.giphyCall.bind(this);
    // this.giphyChange = this.giphyChange.bind(this);
  }
  // Using 'public class fields syntax' to correctly bind callbacks with arrow function, ensuring 'this' is bound within the methods
  // Arrow functions adopt the this binding of the enclosing scope (in other words, they don’t change the meaning of this), so things just work automatically.
  contactBar = () => {
    this.setState((prevState, props) => {
      return { contactVisible: !prevState.contactVisible }
    })
 }

  render() {
    // With JSX you pass a function as the event handler, rather than a string.
    return (
      <main id="main-frame">
      <Routes/>
      <Header/>
      <Footer contactVis={this.state.contactVisible} contactBar={this.contactBar}/>
      </main>
  )
  }
}

export default App;

 /* contactBar={() => this.contactBar()} The problem with this syntax is that a different callback is created each time App renders. if this callback is passed as a prop to lower components, those componets might do extra re-rendering - performance issues */

//   giphyCall = () => {
//  fetch(`https://api.giphy.com/v1/gifs/search?api_key=1407a89874814d21a7d7c728b08138d5&q=${this.state.giphyInput}&limit=30&offset=0&rating=R&lang=en`)
//   .then(res => {
//       return res.json();
//   }).then(data => {
//     this.setState((prevState, props) => {
//       console.log(prevState)
//       console.log(props)
//       return {giphyArr: data.data}
//     })
//   }).catch(err => {
//       console.log(err);
//   });

// // }

// giphyChange = (event) => {
//   this.setState({
//     giphyInput: event.target.value,
//   })
// }
