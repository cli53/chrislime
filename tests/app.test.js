import React from 'react';
import expect from 'expect';
import sinon from 'sinon';
import { shallow, render, mount } from 'enzyme';
import Header from '../src/components/Header.jsx';
import Footer from '../src/components/Footer.jsx';
import App from '../src/App.jsx';
// The shallow function in Enzyme does a shallow rendering to the DOM. Shallow rendering does not render any components nested within the Welcome component.

describe('React unit tests', () => {

  describe('<App/>', () => {
    it('renders 1 <App/> component', () => {
      const wrapper = shallow(<App/>)
      const appState = wrapper.state().contactVisible
      expect(appState).toEqual(false);
    })
  })

  describe('<Footer/>', () => {
    let wrapper;
    let clickSpy;

    it('Changes contactVis to true', () => {
      clickSpy = sinon.spy();
      wrapper = shallow(<Footer contactVis={true} contactBar={clickSpy}/>);
      wrapper.simulate('click');
      expect(wrapper.props().className).toEqual('slideOut');
      expect(wrapper.childAt(0).name()).toBe('p');
      expect(wrapper.childAt(0).text()).toBe('Contact');
    })
  }) 
})