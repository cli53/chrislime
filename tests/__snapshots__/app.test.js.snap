// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests Intro component matches the snapshot 1`] = `
<div
  id="intro"
>
  <p
    className="header-one sec-pad"
  >
    Hi! I'm Chris, Product Engineer based in NYC
    <span
      id="red-dot"
    >
      .
    </span>
  </p>
</div>
`;
